packages/solc-transpiler/src/compiler.ts

```diff
@@ -110,6 +113,53 @@ export const compile = (configJsonString: string, callbacks?: any): string => {
   return formatOutput(res, json)
 }
 
+const getCompiler = (versionString: string): typeof solc => {
```

Comment:
This is awesome. Definitely a shame that we have to do all of this, but it's a great work around.

I'm trying to think if we can write less code that assumes stuff about solc-js in general. There's the github url, and code copied from _this_ version of solc. I wonder if we can turn `getCompiler(...)` into `compile(...)` and it will resolve the compiler _and_ compile all in one. The benefit of that is that in the spawned process you can just 
* import solc
* call solc.loadRemoteVersion(...)
* call `compile(...)` on that version and return the resulting string

This has the benefit of just using the public solc API and not assuming any internals. Are there any downsides? Would error handling be a pain? Unless catastrophic, I think errors are just returned in the JSON.

Reply:
Separately, there's a bit going on here, so I might suggest adding a function header comment.

Reply:
Agreed I'd love to find a way to make use of `solc` and remove those constants if possible!

The issue with calling `compile` in the `execSync` block is that I'm pretty sure callbacks won't work since it's running in a separate node process.  `compile` takes a callback [here](https://github.com/ethereum-optimism/optimism-monorepo/blob/YAS-183/HardenTheOvm/ResearchSolidityTesting/packages/solc-transpiler/src/compiler.ts#L59).
If there was some sort of `solc.loadRemoteVersionSource` function we could make use of that but I don't see one 😕

Reply:
Hmmm... yeah, you're right on that. Really there shouldn't be any callbacks, but there will be functions to resolve library paths eventually. I think I agree with your approach. Will be much better to do as little logic as possible in a separate proc so we can get back to our main proc with better logging and whatnot.